### PR TITLE
Make `v` optional in tag when matching version

### LINKED ISSUE #258: Kebechet version manager should match v* tags
It is common practice to tag releases as `vX.Y` while the version is `X.Y`.
When Kebechet computes changelog between two releases, it matches is to a version... releases tagged `v*` do not match the version exactly and an initial release is assumed.

**Expected behaviour**
The changelog is computed correctly and as a tag `vX.Y` matches version `X.Y`

### PR DESCRIPTION
**Pull request by CermakM**
Fixes #258

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   kebechet/managers/version/version.py
new file:   kebechet/managers/version/version_test.py

### CODE REVIEW

File: kebechet/managers/version/version_test.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+"""Tests for version manager."""
+
+import unittest
```
Comment (fridex) [inline kebechet/managers/version/version_test.py]: In project thoth we adopted `pytest` and `flexmock` for writing and running tests. Could be this test scenario rewritten to those frameworks?

File: kebechet/managers/version/version_test.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        with patch.object(FakeGit, "log") as patch_log:
+            # exact match
+            _ = self.manager._compute_changelog(
+                repo=fake_repo,
+                old_version="0.1.0",
+                new_version="0.2.0",
+                version_file=False,
+            )
+            patch_log.assert_called_with(
+                f"0.1.0..HEAD", no_merges=True, format="* %s"
```
Comment (fridex) [inline kebechet/managers/version/version_test.py]: f-strings not needed in these cases (here and bellow).

Comment (CermakM) [reply kebechet/managers/version/version_test.py]: Ah, correct! Leftover, thanks :+1:

File: kebechet/managers/version/version_test.py (hunk 1 above)
Comment (CermakM) [reply kebechet/managers/version/version_test.py]: Right, ye sure, shouldn't be a problem.

**Comment by goern** [comment]
https://github.com/thoth-station/zuul-config/blob/master/zuul.d/_pipelines.yaml#L116-L152 pre-release and release pipelines are affected by this change

**Comment by CermakM** [comment]
@goern how are they affected?

**Comment by CermakM** [comment]
@fridex I simply wasn't able to use flexmock for that ... I spent quite some time getting various errors from it, so I stuck with the mock and migrated at least the unittest to pytest.

**Comment by goern** [comment]
> @goern how are they affected?

if we start using v-tags the pipeline's regex will not trigger on tag creation, iirc

**Comment by CermakM** [comment]
We don't have to change anything, the `v` is optional in the tag.

**Comment by goern** [comment]
@fridex wdyt?

**Comment by goern** [comment]
@saisankargochhayat @fridex shall we merge this?!

**Comment by fridex** [comment]
> @saisankargochhayat @fridex shall we merge this?!

Do we want to have `v` prefix in versions? Will it help on tekton side?

**Comment by goern** [comment]
Lets do it. @saisankargochhayat can you please validate that all bots and zuul and tekton use a v-prefixed tag on git and quay?

Cc: @harshad16

**Pull request merged**